data-products/src/main/scala/org/sunbird/analytics/sourcing/SourcingMetrics.scala

```diff
@@ -0,0 +1,173 @@
+import org.ekstep.analytics.framework.util.{HTTPClient, JSONUtils, JobLogger, RestUtil}
+import org.ekstep.analytics.framework.{FrameworkContext, IJob, JobConfig, Level, StorageConfig}
+import org.ekstep.analytics.model.ReportConfig
+import org.ekstep.analytics.util.Constants
+import org.sunbird.analytics.job.report.BaseReportsJob
+import org.sunbird.analytics.model.report.{TenantInfo, TenantResponse, TextbookData}
+import org.sunbird.analytics.util.TextBookUtils
+import org.sunbird.cloud.storage.conf.AppConf
+
+case class ContentHierarchy(identifier: String, hierarchy: String)
+case class TextbookReportResult(identifier: String, l1identifier: String,board: String, medium: String, grade: String, subject: String, name: String, chapters: String, channel: String, totalChapters: String)
+case class ContentReportResult(identifier: String, l1identifier: String, contentType: String)
+case class TextbookHierarchy(channel: String, board: String, identifier: String, medium: Object, gradeLevel: List[String], subject: Object,
+                             name: String, status: String, contentType: Option[String], leafNodesCount: Int, lastUpdatedOn: String,
+                             depth: Int, createdOn: String, children: Option[List[TextbookHierarchy]], index: Int, parent: String)
+case class FinalReport(identifier: String,l1identifier: String,board: String, medium: String, grade: String, subject: String, name: String, chapters: String, channel: String, totalChapters: String, slug:String)
+case class TextbookResponse(l1identifier:String,board: String, medium: String, grade: String, subject: String, name: String, chapters: String, channel: String)
+
+object SourcingMetrics extends optional.Application with IJob with BaseReportsJob {
+
+  implicit val className = "org.sunbird.analytics.job.SourcingMetrics"
+  val jobName: String = "Sourcing Metrics Job"
+  val sunbirdHierarchyStore: String = AppConf.getConfig("course.metrics.cassandra.sunbirdHierarchyStore")
+
+  // $COVERAGE-OFF$ Disabling scoverage for main method
+  def main(config: String)(implicit sc: Option[SparkContext], fc: Option[FrameworkContext]): Unit = {
+    JobLogger.log(s"Started execution - $jobName",None, Level.INFO)
+    implicit val sparkContext: SparkContext = getReportingSparkContext(JSONUtils.deserialize[JobConfig](config))
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    val readConsistencyLevel: String = AppConf.getConfig("course.metrics.cassandra.input.consistency")
+    val sparkConf = sparkContext.getConf
+      .set("es.write.operation", "upsert")
+      .set("spark.cassandra.input.consistency.level", readConsistencyLevel)
+    val spark = SparkSession.builder.config(sparkConf).getOrCreate()
+
+    generateSourcingMetrics(spark, config)
+  }
+
+  // $COVERAGE-ON$ Enabling scoverage for all other functions
+  def generateSourcingMetrics(spark: SparkSession, config: String)(implicit sc: SparkContext,fc: FrameworkContext): Unit = {
```

Comment:
Why do we need to pass in both spark and sparkcontext? Pass SparkSession, Config and FC as implicit variables